chromadb/execution/executor/distributed.py

```diff
@@ -112,19 +112,21 @@ def get(self, plan: GetPlan) -> GetResult:
 
     @overrides
     def knn(self, plan: KNNPlan) -> QueryResult:
-        records = self._metadata_segment(plan.scan.collection).get_metadata(
-            request_version_context=plan.scan.version,
-            where=plan.filter.where,
-            where_document=plan.filter.where_document,
-            ids=plan.filter.user_ids,
-            limit=None,
-            offset=0,
-            include_metadata=False,
-        )
+        prefiltered_ids = None
+        if plan.filter.user_ids or plan.filter.where or plan.filter.where_document:
+            records = self._metadata_segment(plan.scan.collection).get_metadata(
+                request_version_context=plan.scan.version,
+                where=plan.filter.where,
+                where_document=plan.filter.where_document,
+                ids=plan.filter.user_ids,
+                limit=None,
+                offset=0,
+                include_metadata=False,
+            )
+            prefiltered_ids = [r["id"] for r in records]
 
-        prefiltered_ids = [r["id"] for r in records]
         knns: Sequence[Sequence[VectorQueryResult]] = [[]] * len(plan.knn.embeddings)
-        if len(prefiltered_ids) > 0:
+        if prefiltered_ids is None or len(prefiltered_ids) > 0:
```

Comment:
Doesn't this query step always run? Is this only avoiding the case where len(prefiltered_ids) == 0? Can we make that more explicit then?

Reply:
Added comments for this